jp/components/grid_groupby.md

```diff
@@ -0,0 +1,140 @@
+_language: ja
+---
+
+### グリッドのグループ化
+
+`igxGrid` は、列グループ値に基づいて階層グループのデータ レコードを可視化できる列グループをサポートします。グループは展開または縮小でき、グループのは UI または API で順序を変更できます。
+
+
+#### デモ
+
+<div class="sample-container loading" style="height:550px">
+    <iframe id="grid-sample-pinning-iframe" src='{environment:demosBaseUrl}/grid-sample-groupby' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<br/>
+<div>
+<button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="grid-sample-groupby-iframe" data-demos-base-url="{environment:demosBaseUrl}">stackblitz で表示</button>
+</div>
+
+#### 初期のグループ化状態
+
+グリッドの初期のグループ化状態を設定するには、式の配列をグリッドの filteringExpressions プロパティに設定します。
+
+```typescript
+public ngOnInit() {
+    grid.groupingExpressions = [
+        { fieldName: 'ProductName', dir: SortingDirection.Desc },
+        { fieldName: 'Released', dir: SortingDirection.Desc }
+    ];
+}
+```
+
+グループ式は、`ISortingExpression` インターフェイスを実装します。
+
+#### グループ化 API
+
+グループ化は、UI およびグリッド コンポーネントで公開された API で実行できます。各列の `groupable` プロパティを `true`.に設定してエンドユーザーは特定の列でグリッド データをグループ化できます。
+
+```html
+<igx-grid [data]="data">
+    <igx-column *ngFor="let c of columns" [field]="c.field" [groupable]="true">
+    </igx-column>
+</igx-grid>
+```
+
+```typescript
+public ngOnInit() {
+    grid.columns.forEach((column) => {
+        column.groupable = true;
+    });
+}
+```
+
+ランタイムの式は `groupingExpressions` プロパティの gettable および settable です。既存の式を追加または変更する必要がある場合、`ISortingExpression` の単一または配列 の `groupBy` メソッドを使用してください。
+
+```typescript
+grid.groupBy({ fieldName: 'ProductName', dir: SortingDirection.Desc, ignoreCase: true });
+```
+
+グループ化は、並べ替えおよびグループ式が列がグループ化されずに並べ替えられた場合も並べ替えされた列として適用される場合に可能です。グループ式は `sortingExpression` プロパティにあり、グループ列の式が常に最初になるよう並べ替えます。並べ替え API は、グループ化された列の式を変更した場合にグルーピングに影響があります。グループ式 (列グループ解除) を削除して並べ替えを解除します。
+
+グループ式の他にグループ行の展開も制御できます。`igxGrid` コンポーネント `groupingExpansionState` の別のプロパティに保存されます。グループ行は、作成されたフィールド名とグループ化の各レベルを表す値に基づいて識別されます。以下は展開状態のインターフェイスのシグネチャです。
+
+```typescript
+export interface IGroupByKey {
+    fieldName: string;
+    value: any;
+}
+
+export interface IGroupByExpandState {
+    hierarchy: Array<IGroupByKey>;
+    expanded: boolean;
+}
+```
+
+`groupingExpression` で `IGroupByExpandState` のリストを直接 `groupingExpansionState` に設定すると展開が変更されます。`igxGrid` はグループ レコード インスタンスでトグルするメソッドを公開します。
+
+```typescript
+    const groupRow = grid.getGroupRow([{ fieldName: 'Category', value: 'soft drinks' }]);
+    grid.toggleGroup(groupRow);
+```
+
+Groups は expanded (***デフォルト***) または collapsed に作成でき、展開状態は一般的にデフォルト動作の反対の状態のみ含みます。グループを作成して展開するかどうか、または `groupsExpanded` プロパティを介すかどうかを制御できます。
```

Comment:
@yoshimis "Groups" should be translated here, as it's a generic term.  If "expanded" and "collapsed" should be kept in English (the source does not specify that they are code words), then add backticks (ex. `expanded`).  Or translate them.

Reply:
Translated.